ddl/partition.go

```diff
@@ -812,6 +812,19 @@ func (w *worker) onExchangeTablePartition(d *ddlCtx, t *meta.Meta, job *model.Jo
 	// exchange table meta id
 	partDef.ID = nt.ID
 
+	// Clear the tiflash replica available status.
+	if pt.TiFlashReplica != nil {
+		pt.TiFlashReplica.Available = false
```

Comment:
why set `Available` to false? I think if exchange partition is allowed, the TiFlash replica status of both table are  always the same(both available or both not available), it is not necessary to reset the TiFlash replica status to false if they are already available.

Reply:
One more question: is it possible that the `available` flag is changed between `checkTiFlashReplicaCompatible ` and `onExchangeTablePartition `, if yes, then we need handle it more carefully.

Reply:
I get that It seem that `tiflash`  now doesnot support exchange partition in slack. So I just reset false like truncate partition.  I would delete the code about changing `tiflash` status.

Reply:
TiFlash does not support exchange partition when #17149 is merged, and it has supported now, so you can safely delete related code, but take care of my last question, if `available` flag is changed, we need to handle the status carefully for both partition table and non-partition table.

Reply:
I hava a question about how to change a `available` status when a `exchange partition` happening ? It seems that only one worker handle ddl events.

Reply:
> I hava a question about how to change a `available` status when a `exchange partition` happening ? It seems that only one worker handle ddl events.

I'm not very familiar with TiDB, if `available` status can not be changed during `exchange partition`, then just ignore my question.